src/App.tsx

```diff
@@ -14,11 +15,107 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  list: string[]
+  sortBy: string
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    list: [],
+    sortBy: '',
+  };
+
+  showList = () => {
+    this.setState({ list: [...goodsFromServer] });
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      list: state.list.reverse(),
+      sortBy: '',
+    }));
+  };
+
+  sortByAZ = () => {
+    this.setState({ sortBy: 'AZ' });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortBy: 'length' });
+  };
+
+  reset = () => {
+    this.setState({ list: [...goodsFromServer], sortBy: '' });
+  };
+
+  render() {
+    const { list = [''], sortBy } = this.state;
+
+    list.sort((g1, g2) => {
+      switch (sortBy) {
+        case 'AZ':
+          return g1.localeCompare(g2);
+
+        case 'length':
+          return g1.length - g2.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    return (
+      <div className="App">
+        <button
+          type="button"
+          className="button"
+          onClick={this.showList}
+          style={{ display: (list.length > 0) ? 'none' : 'visible' }}
+        >
+          Start
+        </button>
+
+        <RenderList list={this.state.list} />
```

Comment:
```suggestion
        <GoodsList list={this.state.list} />
```
RenderList звучит как название функции, а не как компонент